coordinator/src/test/scala/filodb.coordinator/queryplanner/SingleClusterPlannerSpec.scala

```diff
@@ -253,4 +255,50 @@ class SingleClusterPlannerSpec extends FunSpec with Matchers {
     }
   }
 
+  it("should bound queries until retention period and drop instants outside retention period") {
+     val planner = new SingleClusterPlanner(dsRef, schemas, mapperRef,
+       earliestRetainedTimestampFn = System.currentTimeMillis - 3.days.toMillis)
+
+    val nowSeconds = System.currentTimeMillis() / 1000
+
+    // Case 1: no offset or window
+    val logicalPlan1 = Parser.queryRangeToLogicalPlan("""foo{job="bar"}""",
+      TimeStepParams(nowSeconds - 4.days.toSeconds, 1.minute.toSeconds, nowSeconds))
+
+    val ep1 = planner.materialize(logicalPlan1, QueryContext()).asInstanceOf[DistConcatExec]
+    val psm1 = ep1.children.head.asInstanceOf[MultiSchemaPartitionsExec]
+                .rangeVectorTransformers.head.asInstanceOf[PeriodicSamplesMapper]
+    psm1.start shouldEqual (nowSeconds * 1000
+                            - 3.days.toMillis // retention
+                            + 1.minute.toMillis // step
+                            + WindowConstants.staleDataLookbackSeconds * 1000) // default window
```

Comment:
Why should staleDataLookbackSeconds be added?

Reply:
This is because the query `foo{job="bar"}` does not have a lookback window, and in this case, staleDataLookbackSeconds comes in.